### PR TITLE
Update rollup plugins and fix their ordering #105

### PR DESCRIPTION
**Pull request by jcoussard**
Only dev related changes. No need for a new version release.

### CODE REVIEW

**Comment by jcoussard** [comment]
This pull request fixes the bundles to inline the local code dependencies instead of importing it. This was noticed after trying to build the project and noticing the .size-snapshot.json was way different.

The first attempts at fixing it were suggested by @dai-shi. I switched the following rollup packages:
- rollup-plugin-node-resolve => @rollup/plugin-node-resolve
- rollup-plugin-typescript2 => @rollup/plugin-typescript

This didn't solve the root issue, but instead resulted in new issues. The @rollup/plugin-typescript didn't like the fact that declarations were outputted without providing an "outDir" value in tsconfig. But if specifying one, then new problems arose in the existing rollup bundles.

After looking for ideas from other projects, I noticed that the resolve() plugin was always put first in all rollup configurations, I tried it and it worked! Then all was left was to re-emit the typescript declaration files. Since I thought it was a good idea to keep the updated dependencies, I created a new rollup entry specifically to emit them.

**Comment by jcoussard** [comment]
Also I guess zustand could profit from the same fix: https://github.com/pmndrs/zustand/blob/master/rollup.config.js

File: rollup.config.js [hunk 1]
```diff
@@ -14,16 +14,27 @@ const getBabelOptions = (targets) => ({
...
 })
 
+function createDeclarationConfig(input, output) {
+  return {
+    input,
+    output: {
+      dir: output,
+    },
+    external,
+    plugins: [typescript({ declaration: true, outDir: output })],
```
Comment (dai-shi) [inline rollup.config.js:24]: I tried almost all what you tried, and agree this would be the best so far. 😄

File: package.json [hunk 2]
```diff
@@ -117,9 +119,7 @@
     "react-dom": "^17.0.0",
     "rollup": "^2.32.0",
     "rollup-plugin-babel": "^4.4.0",
```
Comment (dai-shi) [inline package.json:121]: Would you please try https://www.npmjs.com/package/@rollup/plugin-babel too?

**Comment by dai-shi** [comment]
Thanks for your investigation!

> Also I guess zustand could profit from the same fix

I agree. Would you be interested to work on it too? Otherwise, I can ask someone for help in the discord channel.

File: package.json (hunk 2 above)
Comment (jcoussard) [reply package.json:121]: I thought the same while going to bed last night. I'll try it later today.

**Review by dai-shi — APPROVED** [review]
LGTM. Thanks for working on this.

**Pull request merged**